src/rpc/misc.cpp

```diff
@@ -301,16 +275,41 @@ UniValue createmultisig(const JSONRPCRequest& request)
             "}\n"
 
             "\nExamples:\n"
-            "\nCreate a multisig address from 2 addresses\n"
-            + HelpExampleCli("createmultisig", "2 \"[\\\"16sSauSf5pF2UkUwvKGq4qjNRzBZYqgEL5\\\",\\\"171sgjn4YtPu27adkKGrdDwzRTxnRkBfKV\\\"]\"") +
+            "\nCreate a multisig address from 2 public keys\n"
+            + HelpExampleCli("createmultisig", "2 \"[\\\"03789ed0bb717d88f7d321a368d905e7430207ebbd82bd342cf11ae157a7ace5fd\\\",\\\"03dbc6764b8884a92e871274b87583e6d5c2a58819473e17e107ef3f6aa5a61626\\\"]\"") +
             "\nAs a json rpc call\n"
-            + HelpExampleRpc("createmultisig", "2, \"[\\\"16sSauSf5pF2UkUwvKGq4qjNRzBZYqgEL5\\\",\\\"171sgjn4YtPu27adkKGrdDwzRTxnRkBfKV\\\"]\"")
+            + HelpExampleRpc("createmultisig", "2, \"[\\\"03789ed0bb717d88f7d321a368d905e7430207ebbd82bd342cf11ae157a7ace5fd\\\",\\\"03dbc6764b8884a92e871274b87583e6d5c2a58819473e17e107ef3f6aa5a61626\\\"]\"")
         ;
         throw std::runtime_error(msg);
     }
 
+    int required = request.params[0].get_int();
+
+    // Get the public keys
+    const UniValue& keys = request.params[1].get_array();
+    std::vector<CPubKey> pubkeys;
+    pubkeys.resize(keys.size());
+    for (unsigned int i = 0; i < keys.size(); ++i) {
+#ifdef ENABLE_WALLET
+        if (IsDeprecatedRPCEnabled("createmultisig")) {
+            CWallet* const pwallet = GetWalletForJSONRPCRequest(request);
+            if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
```

Comment:
There's an implicit assumption here that if `ENABLE_WALLET` is true then there will necessarily be a wallet available. That may be broken by dynamic wallet load/unload. If we allow the wallet to be unloaded then this rpc will stop working.

The else statement is also unnecessary (once we reach line 302, then both branches converge to calling `_createmultisig_hex_to_pubkey()`). How about

How about:

```
#ifdef ENABLE_WALLET
        CWallet* const pwallet = GetWalletForJSONRPCRequest(request);
        if (IsDeprecatedRPCEnabled("createmultisig") &&
            EnsureWalletIsAvailable(pwallet, request.fHelp) &&
            pwallet) {
            _createmultisig_addr_to_pubkey(pwallet, pubkeys.at(i), keys[i].get_str());
        }
#endif
```

Reply:
Done.